glean-core/ios/Glean/Metrics/LabeledMetric.swift

```diff
@@ -0,0 +1,122 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+typealias LabeledMetricCtor = (FfiStr?, FfiStr?, RawStringArray?, Int32, Int32, UInt8, RawStringArray?, Int32) -> UInt64
+
+/// This implements the developer facing API for labeled metrics.
+///
+/// Instances of this class type are automatically generated by the parsers at build time,
+/// allowing developers to record values that were previously registered in the metrics.yaml file.
+///
+/// Unlike most metric types, LabeledMetricType does not have its own corresponding storage engine,
+/// but records metrics for the underlying metric type `T` in the storage engine for that type.
+/// The only difference is that labeled metrics are stored with the special key `$category.$name/$label`.
+public class LabeledMetricType<T> {
```

Comment:
Would you kindly add the docs for the labeled metric types, e.g. [`labeled_counter`](https://mozilla.github.io/glean/book/user/metrics/labeled_counters.html)?